student/src/main/java/tds/student/web/handlers/MasterShellHandler.java

```diff
@@ -136,6 +137,9 @@
   private ITDSLogger             _tdsLogger;
 
   @Autowired
+  private StudentEventLogger     _eventLogger;
```

Comment:
The StudentEventLogger class is annotated with a magic Spring 'scoped proxy mode' which arranges for a new instance to be created for each request (I tested it by logging from the constructor, and indeed a new instance is made for every single request).

That's in StudentEventLogger.java:12:
`@Scope(value = "request", proxyMode = ScopedProxyMode.TARGET_CLASS)`